### ISSUE TITLE
col_datetime issues with ISO dates

### ISSUE THREAD

**Issue by thrasibule** [hadley/readr#180]
```R
test<-"date,val\n2015-05-22 10:36:27,0.5"
data<-read_csv(test, col_names=T, list(col_datetime(), col_double()))
data<-read_csv(test, col_names=T, list(col_datetime("%F %T"), col_double()))
data<-read_csv(test, col_names=T, list(col_datetime("%Y-%m-%d %T"), col_double()))
```
My understanding is those three ways of reading the file should be equivalent. However only the first one works. The second one parses the date as "2015-01-01 10:36:27", and the second one doesn't parse it at all. This happens with latest version of readr from github, as well as the released version in CRAN.

**Comment by hadley** [comment]
I think this is fixed by #178

**Comment by thrasibule** [comment]
Yes it does fix it indeed.
